app/geckoModule/controller.js

```diff
@@ -73,6 +73,9 @@ class MainController {
         }
 
         const audio = urlParams.get('audio')
+        
+        const presignedSplited = window.location.search.split('presigned_url=')
```

Comment:
Use url params here.
Assume that the links in the url params are escaped.
Please check for any other place in the code and adjust accordingly